packages/@vuepress/theme-default/layouts/Layout.vue

```diff
@@ -21,11 +21,11 @@
     >
       <slot
         name="sidebar-top"
-        #top
+        slot="top"
       />
```

Comment:
The new `v-slot` syntax can only be added to a `<template>`, so the code below should fix it
```
<template #top>
  <slot name="sidebar-top"/>
</template>
```